lib/src/query.c

```diff
@@ -3758,6 +3767,9 @@ static inline bool ts_query_cursor__advance(
             }
             if (!has_supertype) node_does_match = false;
           }
+          if (step->is_missing) {
+            node_does_match = is_missing;
+          }
```

Comment:
I would think this should be higher up, because this would overwrite this being false in case there's later siblings and this is supposed to be the last child. I would put it inside the if/else check of the wildcard symbol, basically adding the missing part onto the conditional in each branch.